### PR TITLE
[UPDATES] read me

### PR DESCRIPTION
**Pull request by MWahhab**
README.md:
- Now includes comprehensive steps for both Windows and WSL/Linux users.
- Utilises a more granular approach to explain how to interact with the containers through the docker desktop, in order to execute the necessary steps to setup the project.
- Includes directions on how to retrieve .env.example.
- Includes alternate methods that the user can use to access the application; with one such step being through the nginx container on docker desktop.
- Depicts how to execute scripts using php-cli, as well as artisan commands through php-fpm.
- Includes more whitespaces/linebreaks to create a more visually pleasing and less cluttered documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,47 +13,130 @@ Before proceeding, make sure you have installed:
```
Comment (lolmeherti) [inline README.md:14]: Can you add git to Prerequisites?

Comment (MWahhab) [reply README.md:14]: lemme think about it and ill get back to you

File: README.md [hunk 2]
```diff
@@ -13,47 +13,130 @@ Before proceeding, make sure you have installed:
...
-3. **Build and Run with Docker Compose**
-
-    Build the Docker images and start the services:
-
-    ```bash
-    docker-compose up --build -d
+   <br/>
+3. **Set Up Environment File**
+
+    Copy the `.env.example` file from the official Laravel github repository to `.env`. Then fill in the database
```
Comment (lolmeherti) [inline README.md]: Copy it where? The dockerizedLaravel skeleton also comes with a laravel application inside of it by default. We don't mention that here, why?

File: README.md [hunk 3]
```diff
@@ -13,47 +13,130 @@ Before proceeding, make sure you have installed:
...
+   <br/>  
+   
+   ***If you have attempted to build the containers previously, execute the following to rebuild them***
+   ```bash
+   docker-compose up -d --force-recreate
+   ```
+   <br/>
+6. **Install Composer Packages**  
+   Navigate to the `docker desktop` - then select the running ``php-cli`` container. Click on the `exec` tab. 
+   Complete the following in the command line:<br/>
```
Comment (lolmeherti) [inline README.md:169]: Can you add instructions for non docker desktop users?

File: README.md [hunk 4]
```diff
@@ -13,47 +13,130 @@ Before proceeding, make sure you have installed:
...
+   ```bash
+   composer -v
+   ```
+   ***If it isn't installed, ensure that the image has been built correctly; has the correct pathing.***<br/>  
+   If there's no issues, install the packages
+   ```bash
+   composer install
+   ```
+   <br/>
+7. **Generate API key**
```
Comment (lolmeherti) [inline README.md]: I think you mean the application key?

File: README.md [hunk 5]
```diff
@@ -13,47 +13,130 @@ Before proceeding, make sure you have installed:
...
+   cd laravel
+   ```
+   ***If the path doesn't exist, ensure that the pathing for the mount of the `php-fpm` container is correct***
+   <br/>   
+   Generate API key
+   ```bash
+   php artisan key:generate
+   ```
+   <br/>
+8. **Run Migrations (Optional)**
```
Comment (lolmeherti) [inline README.md]: Not really optional as laravel by default always has at least a few migration files.

**Review by lolmeherti — CHANGES_REQUESTED** [review]
Please check the comments.

File: README.md [hunk 6]
```diff
@@ -8,52 +8,226 @@ Before proceeding, make sure you have installed:
...
+   ***If it isn't installed, ensure that the image has been built correctly; has the correct pathing.***<br/>  
+   If there's no issues, install the packages
+   ```bash
+   composer install
+   ```
+   ---
+   **IDE**
+   <br/>  
+   In the CMD, navigate to the application directory (`src/laravel`) - then execute the following:
+   <br/>
```
Comment (lolmeherti) [inline README.md]: Can you add a bit more detail on commands? Do docker ps to see containers running, docker exec -it on the bash terminal.

**Review by lolmeherti — CHANGES_REQUESTED** [review]

**Pull request merged**